generators/typescript-v2/ast/src/ast/Func.ts

```diff
@@ -0,0 +1,89 @@
+import { CodeBlock } from "./CodeBlock";
+import { Comment } from "./Comment";
+import { Parameter } from "./Parameter";
+import { Type } from "./Type";
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+
+export declare namespace Func {
+    interface Args {
+        name: string;
+        parameters: Parameter[];
+        return_: Type[];
+        body: CodeBlock | undefined;
+        docs: string | undefined;
+    }
+}
+
+export class Func extends AstNode {
```

Comment:
nit: Mind renaming this to `Function`? We try to keep the AST nodes named based on the language. For Go we refer to them as `Func`, but for TypeScript we can say `Function`.

Reply:
Sure, can do